crates/fuel-core/src/service/adapters/consensus_module.rs

```diff
@@ -45,7 +48,7 @@ impl VerifierAdapter {
     }
 }
 
-impl fuel_core_poa::ports::Database for Database {
+impl fuel_core_poa::ports::Database for OnChainIterableView {
     fn block_header(&self, height: &BlockHeight) -> StorageResult<BlockHeader> {
         Ok(self.get_block(height)?.header().clone())
```

Comment:
I guess I'm confused why we are calling the database we use here `Iterable`. Since this doesn't really iterate.

Reply:
Well, `IterableStore::iter_store()` returns a `BoxedIter<KVItem>`. So that implies sometimes its purpose is iterating.

`BasicView` isn't bad and we can depend on `IterableStore` where we need to with an `impl IterableStore for BasicView` maybe?

Reply:
The idea of `BasicView` is that it doesn't implement `IterableStore`. Only `View`(`IterableView`) will implement `IterableStore`.

Reply:
Oh. Add another view?

Reply:
I lost the line of conversation=D The `IterableView` is an extension of the `KeyValueView`. It is the same as `KeyValueView` but also allows iteration of the storage.

You don't like the name `IterableView`. I'm proposing to change it to `View` and change the name of `KeyValueView` to `BasicView`. Where is the third view here?=D

Reply:
That was the missing piece for my understanding. I get it now. 

In that case, I'm in favor of `IterableKeyValueView`. It's verbose but very clear.